syft/pkg/cataloger/javascript/parse_package_json.go

```diff
@@ -64,11 +63,8 @@ func parsePackageJSON(_ context.Context, _ file.Resolver, _ *generic.Environment
 			return nil, nil, fmt.Errorf("failed to parse package.json file: %w", err)
 		}
 
-		if !p.hasNameAndVersionValues() {
-			log.Debugf("encountered package.json file without a name and/or version field, ignoring (path=%q)", reader.Path())
-			return nil, nil, nil
-		}
-
+		// always create a package, regardless of having a valid name and/or version,
+		// a compliance filter later will remove these packages based on compliance rules
```

Comment:
This is a small, but potentially impactful change and should be scrutinized. This will result in the cataloger returning packages with empty name and/or versions, whereas these were being entirely dropped before. There are quite a lot of these files in the sample set of image I tested on that have no name. These will be filtered out AFTER being returned by compliance rules. There will be a number of packages NOT filtered out that do not have versions, however, which will decidedly be a change that some users notice.